release/src/module_dependencies/main.py

```diff
@@ -0,0 +1,172 @@
+        processedLine = line.decode("utf-8")
+
+        if 'ballerina-platform/module' in processedLine:
+            module = processedLine.split('/')[-1]
+            if module[:-2] == balModule:
+                continue
+            dependencies.append(module[:-2])
+
+    return dependencies
+
+# Gets the version of the ballerina standard library module from gradle.properties file in module repository
+# returns: current version of the module
+def getVersion(balModule):
+
+    data = urllib.request.urlopen("https://raw.githubusercontent.com/ballerina-platform/" + balModule + "/master/gradle.properties")
+
+    for line in data:
+
+        processedLine = line.decode("utf-8")
+
+        if 'version' in processedLine:
+            version = processedLine.split('=')[-1][:-1]
+
+    return version 
+
+# access the module_list.json file
+# returns: a list containing an array of standard library module names
+def getModuleNameList():
+    with open('./release/resources/module_list.json') as f:
+          fileContent = json.load(f)
+
+    return fileContent['modules']
+
+# Calculates the longest path between source and destination modules and replaces dependents that have intermediates
+def removeModulesInIntermediatePaths(G, source, destination, successors, moduleDetailsJSON):
+
+    longestPath = max(nx.all_simple_paths(G, source, destination), key=lambda x: len(x))
+
+    for n in longestPath[1:-1]:
+        if n in successors:
+            for module in moduleDetailsJSON['modules']:
+                if module['name'] == source:
+                    if destination in module['dependents']:
+                        module['dependents'].remove(destination)
+                    break
+
+# Generates a directed graph using the dependencies of the modules
+# Level of each module is calculated by traversing the graph 
+# Returns a json string with updated level of each module
+def calculateLevels(moduleNameList, moduleDetailsJSON):
+
+    G = nx.DiGraph()
+
+    # Module names are used to create the nodes and the level attribute of the node is initialized to 0
+    for module in moduleNameList:
+        G.add_node(module, level=0)
+
+    # Edges are created considering the dependents of each module
+    for module in moduleDetailsJSON['modules']:
+        for dependent in module['dependents']:
+            G.add_edge(module['name'], dependent)
+
+    processingList = []
+
+    # Nodes with no in degrees = 0 and out degrees != 0 are marked as level 1 and the node is appended to the processing list
+    for root in [node for node in G if G.in_degree(node) == 0 and G.out_degree(node) != 0]:
+        G.nodes[root]['level'] = 1
+        processingList.append(root)
+
+    # While the processing list is not empty, successors of each node in the current level are determined
+    # For each successor of the node, 
+    # 			longest path from node to successor is considered and intermediate nodes are removed from dependent list - removeModulesInIntermediatePaths
```

Comment:
Check the line length, whether it exceeds 120 characters. Check in the other places as well.
(Do we need the indentation here?)

Let's use spaces instead of tabs.

Reply:
Fixed